Realm/Realm/DatabaseTypes/Accessors/IRealmAccessor.cs

```diff
@@ -0,0 +1,69 @@
+﻿////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Realms.Schema;
+
+namespace Realms
+{
+    public interface IRealmAccessor
+    {
+        bool IsManaged { get; }
+
+        bool IsValid { get; }
+
+        bool IsFrozen { get; }
+
+        Realm Realm { get; }
+
+        ObjectSchema ObjectSchema { get; }
+
+        int? HashCode { get; }
+
+        int BacklinksCount { get; }
+
+        RealmObjectBase.Dynamic DynamicApi { get; }
+
+        RealmValue GetValue(string propertyName);
+
+        void SetValue(string propertyName, RealmValue val);
+
+        void SetValueUnique(string propertyName, RealmValue val);
+
+        IList<T> GetListValue<T>(string propertyName);
+
+        ISet<T> GetSetValue<T>(string propertyName);
+
+        IDictionary<string, TValue> GetDictionaryValue<TValue>(string propertyName);
+
+        IQueryable<dynamic> GetBacklinks(string objectType, string propertyName);
```

Comment:
I suppose I remove it from the interface and implement it only on the ManagedAccessor, so it's not visible

Reply:
Yeah - we probably can move some stuff to `ManagedAccessor` but also rather than return `IQueryable<dynamic>` we can return `IQueryable<IRealmObjectBase>`.